runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1670,13 +1670,22 @@ TR::Node * J9::TransformUtil::calculateOffsetFromIndexInContiguousArray(TR::Comp
    TR::ILOpCodes constOp = comp->target().is64Bit() ? TR::lconst : TR::iconst;
    TR::ILOpCodes shlOp = comp->target().is64Bit() ? TR::lshl : TR::ishl;
    TR::ILOpCodes addOp = comp->target().is64Bit() ? TR::ladd : TR::iadd;
+   TR::ILOpCodes mulOp = comp->target().is64Bit() ? TR::lmul : TR::imul;
 
    if (shift)
       {
       TR::Node *shiftNode = TR::Node::create(TR::iconst, 0);
       shiftNode->setConstValue(shift);
       offset = TR::Node::create(shlOp, 2, offset, shiftNode);
       }
+
+   if (elementStride != 0)
+      {
+      TR::Node *elementStrideNode = TR::Node::create(constOp, 0);
+      elementStrideNode->setConstValue(elementStride);
+      offset = TR::Node::create(mulOp, 2, offset, elementStrideNode);
+      }
+
```

Comment:
You might want to consider checking whether the stride is a power of two and using a shift if it is.

Reply:
Addressed in cf9ae54